test/functional/ws/transport/client.js

```diff
@@ -36,7 +39,27 @@ describe('RPC Client', () => {
 			debug: sinonSandbox.stub(),
 			trace: sinonSandbox.stub(),
 		};
-		validClientRPCStub = connect({ ip, wsPort }, loggerMock).rpc;
+		validPeerStub = connect({ ip, wsPort }, loggerMock);
+		validClientRPCStub = validPeerStub.rpc;
+	}
+
+	function unbindConnectionHandlers() {
+		validPeerStub.socket.removeAllListeners('close');
+		validPeerStub.socket.removeAllListeners('connect');
+	}
+
+	function captureConnectionResult() {
+		closeErrorCode = null;
+		closeErrorReason = null;
+		unbindConnectionHandlers();
+		validPeerStub.socket.on('close', (code, reason) => {
+			unbindConnectionHandlers();
```

Comment:
Why we need that here if we're already calling it 2 lines above?

Reply:
The second one happens asynchronously when the socket disconnects. If the user calls the function twice before the socket closes, then it will remove the previous event listeners.